src/Voron/Impl/FreeSpace/StreamBitArray.cs

```diff
@@ -110,6 +114,180 @@ public bool Get(int index)
+                    {
+                        int current = _inner[i];
+                        if (current == 0)
+                        {
+                            start = -1;
+                            count = 0;
+                            continue;
+                        }
+
+                        if (current == -1)
+                        {
+                            if (start == -1)
+                            {
+                                start = i * BitsInItem;
+                            }
+
+                            count += BitsInItem;
+                            if (count >= num)
+                                return start;
+
+                            continue;
+                        }
+
+                        if (start == -1)
+                        {
+                            // find trailing ones at the end of the block if no sequence has started
+                            CheckTrailingSequence();
+                        }
+                        else
+                        {
+                            if (count + (TotalBits - i * BitsInItem) < num)
+                            {
+                                // impossible to satisfy the continuous bit requirement
+                                return null;
+                            }
+
+                            if (count + 31 < num)
+                            {
+                                // impossible to satisfy the continuous bit requirement in this block
+                                CheckTrailingSequence();
+                                continue;
+                            }
+
+                            // we look at the beginning of the block
+                            int numberOfSetBits = BitOperations.TrailingZeroCount(~current);
+                            count += numberOfSetBits;
+                            if (count >= num)
+                                return start;
+
+                            // reset for the next sequence
+                            CheckTrailingSequence();
+                        }
+
+                        void CheckTrailingSequence()
+                        {
+                            int numberOfSetBits = BitOperations.LeadingZeroCount((uint)~current);
+                            if (numberOfSetBits == 0)
+                            {
+                                start = -1;
+                                count = 0;
+                            }
+                            else
+                            {
+                                // Calculate the starting bit position in the array
+                                start = (i * BitsInItem) + (BitsInItem - numberOfSetBits);
+                                count = numberOfSetBits;
+                            }
+                        }
+                    }
+
+                    return null;
+            }
+        }
+
+        public int? GetContinuousRangeStartLegacy(int num)
```

Comment:
Do we need this?

Reply:
That should go in the test class itself, this is the gold standard (the reference) for a fuzzying test.